lib/screens/explore_screens/flag/interactor/viewmodels/flag_bloc.dart

```diff
@@ -16,11 +19,18 @@ part 'flag_state.dart';
 class FlagBloc extends Bloc<FlagEvent, FlagState> {
   FlagBloc() : super(FlagState.initial()) {
     on<LoadUsersFlags>(_loadUsersFlags);
+    on<OnRemoveUserFlagTapped>(_removeUserFlag);
   }
 
-  Future<FutureOr<void>> _loadUsersFlags(event, Emitter<FlagState> emit) async {
+  FutureOr<void> _loadUsersFlags(event, Emitter<FlagState> emit) async {
```

Comment:
`FutureOr` ?? what is the difference? Should we use this instead `Future<void>`?

Reply:
/// A type representing values that are either `Future<T>` or `T`.
///
/// This class declaration is a public stand-in for an internal
/// future-or-value generic type, which is not a class type.
/// References to this class are resolved to the internal type.
///
/// It is a compile-time error for any class to extend, mix in or implement
/// `FutureOr`.
///
/// # Examples
/// ```dart
/// // The `Future<T>.then` function takes a callback [f] that returns either
/// // an `S` or a `Future<S>`.
/// Future<S> then<S>(FutureOr<S> f(T x), ...);
///
/// // `Completer<T>.complete` takes either a `T` or `Future<T>`.
/// void complete(FutureOr<T> value);
/// ```
///
/// # Advanced
/// The `FutureOr<int>` type is actually the "type union" of the types `int` and
/// `Future<int>`. This type union is defined in such a way that
/// `FutureOr<Object>` is both a super- and sub-type of `Object` (sub-type
/// because `Object` is one of the types of the union, super-type because
/// `Object` is a super-type of both of the types of the union). Together it
/// means that `FutureOr<Object>` is equivalent to `Object`.
///
/// As a corollary, `FutureOr<Object>` is equivalent to
/// `FutureOr<FutureOr<Object>>`, `FutureOr<Future<Object>>` is equivalent to
/// `Future<Object>`.

Reply:
i dont think i makes a difference here. 
But it could be useful in other cases

Reply:
I guess it is like `Result` interface you can get either a Value or Error results.
You can get a Future<int> or a int
`FutureOr<Future<int>, int>`